src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+        newArray.push({ ...copyState });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete copyState[key];
+        }
+        newArray.push({ ...copyState });
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(copyState)) {
+          delete copyState[key];
+        }
+        newArray.push({ ...copyState });
+        break;
+    }
```

Comment:
```suggestion
        break;

      case 'removeProperties':
        for (const key of action.keysToRemove) {
          delete copyState[key];
        }
        break;

      case 'clear':
        for (const key of Object.keys(copyState)) {
          delete copyState[key];
        }
        break;
    }
    
    newArray.push({ ...copyState });
```
you can push to newArray after switch case. it will reduce amount of code.
also add default case (for example throw some errow for unexpected type)